sbi/src/tsm.rs

```diff
@@ -171,6 +171,85 @@ impl TvmCpuRegister {
     }
 }
 
+/// Identifies a register set in the vCPU shared-memory state area layout.
+#[repr(u16)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum RegisterSetId {
+    /// General purpose registers.
+    Gprs = 0,
+    /// HS-level CSRs related to trap handling.
+    HypervisorTrapCsrs = 2,
+}
+
+impl RegisterSetId {
+    /// Returns the `RegisterSetId` corresponding to `val`.
+    pub fn from_raw(val: u16) -> Result<Self> {
+        match val {
+            0 => Ok(RegisterSetId::Gprs),
+            1 => Ok(RegisterSetId::HypervisorTrapCsrs),
+            _ => Err(Error::InvalidParam),
+        }
+    }
+}
+
+/// Specifies the location of a particular register set in the vCPU shared-memory state area.
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct RegisterSetLocation {
+    /// The type of the register set, identified by the `RegisterSetId` enum.
+    pub id: u16,
+    /// The version of the register set structure.
+    pub version: u16,
```

Comment:
Just additional clarification for the documentation: the layout if each register set(e.g.) will be defined in the specification with a version. The version allows us to modify the register set layout in the future.

Reply:
Expanded the comment to include this.